iothub/device/src/ClientFactory.cs

```diff
@@ -22,9 +21,22 @@ internal static class ClientFactory
         {
             Argument.AssertNotNullOrWhiteSpace(connectionString, nameof(connectionString));
 
-            var builder = new IotHubConnectionStringBuilder(connectionString);
+            var csBuilder = new IotHubConnectionStringBuilder(connectionString);
+            if (csBuilder.UsingX509Cert)
+            {
+                throw new ArgumentException("To use X509 certificates, use the initializer with the IAuthenticationMethod parameter.", nameof(connectionString));
+            }
+
+            // Clients that derive their authentication method from AuthenticationWithTokenRefresh will need to specify
+            // the token time to live and renewal buffer values through the corresponding AuthenticationWithTokenRefresh
+            // implementation constructors instead.
+            if (csBuilder.AuthenticationMethod is not AuthenticationWithTokenRefresh)
+            {
+                csBuilder.SasTokenTimeToLive = options?.SasTokenTimeToLive ?? default;
+                csBuilder.SasTokenRenewalBuffer = options?.SasTokenRenewalBuffer ?? default;
```

Comment:
I mostly copied this from below. Not sure how I feel about the TTL default being default(TimeSpan). Probably we should have defaults we apply, but no sure where that would be done. Maybe in `ClientOptions` itself since we have other defaults there too?